### PR TITLE
create token on master, and check for empty master endpoint

### PR DESCRIPTION
**Pull request by oneilcin**
WIP - Please see my TODO comments, it's pretty rough but working.  I'd like to have David's PR 69 merged first as well.

### CODE REVIEW

File: cloud/ssh/actuators/machine/actuator.go [hunk 1]
```diff
@@ -148,6 +148,14 @@ func (a *Actuator) Create(c *clusterv1.Cluster, m *clusterv1.Machine) error {
 
 	glog.Infof("Annotating machine %s for cluster %s.", m.Name, c.Name)
 
+	// TODO remove this when PR is ready https://github.com/samsung-cnct/cluster-api-provider-ssh/pull/69
+	if util.IsMaster(m) {
+		err = a.updateClusterObjectEndpoint(c, m)
```
Comment (davidewatson) [inline cloud/ssh/actuators/machine/actuator.go]: This function updates the resource. Then, just below, the resource is updated again with `a.updateAnnotations(c, m)`. Each time it passes the current object `m`. The second time there is an error:

```
E0824 14:56:44.397208       1 controller.go:105] Error deleting machine object ssh-node-ldm59; Operation cannot be fulfilled on machines.cluster.k8s.io "ssh-node-ldm59": the object has been modified; please apply your changes to the latest version and try again
```

The reason is the resource stored in `etcd` has changed but `m` has not. This is similar to the problem with  https://github.com/samsung-cnct/cluster-api-provider-ssh/pull/73

Comment (davidewatson) [reply cloud/ssh/actuators/machine/actuator.go]: Now I notice that the updates are for different objects. The cluster in the first case and the machine in the second. Hmm.

**Comment by oneilcin** [comment]
We could merge this, it is ready to try out.  We will need to build images based on this for integration testing with the CMA.

**Comment by oneilcin** [comment]
There are still TODOs that can be done later

**Pull request merged**